src/services/model-form-builder.service.ts

```diff
@@ -15,45 +15,33 @@ export class ModelFormBuilder<T> {
       fg.addControl(propertyKey, new FormControl(target[propertyKey]));
       const validators: ValidatorFn[] = [];
 
-      for (const decorator of decorators) {
-
-        switch (decorator) {
-          case 'required': {
-            validators.push(Validators.required);
-            break;
-          }
-          case 'email': {
-            validators.push(Validators.email);
-            break;
-          }
-          case 'min': {
-            const min: number = Reflect.getMetadata(decorator, target, propertyKey);
-            validators.push(Validators.min(min));
-            break;
-          }
-          case 'max': {
-            const max: number = Reflect.getMetadata(decorator, target, propertyKey);
-            validators.push(Validators.max(max));
-            break;
-          }
-          case 'pattern': {
-            const pattern: string | RegExp = Reflect.getMetadata(decorator, target, propertyKey);
-            validators.push(Validators.pattern(pattern));
-            break;
-          }
-          case 'customValidator': {
-            const customValidator: ValidatorFn = Reflect.getMetadata(decorator, target, propertyKey);
-            validators.push(customValidator);
-            break;
-          }
-        }
-
-      }
+      decorators.map((decorator)=>{
```

Comment:
Use a filter to exclude the properties annoted with the metadata **design:type**